controllers/articles-controller.js

```diff
@@ -13,7 +15,39 @@ function getArticleById(req, res, next) {
 }
 
 function getArticles(req, res, next) {
-  selectArticles()
+  selectTopics()
+    .then((topics) => {
+      return Promise.resolve(
+        topics.map((topic) => {
+          return topic.slug;
+        })
```

Comment:
Interestingly here I think you could simplify things, if the `selectTopics()` doesn't return a _Promise.reject_ , can we just assume it must be a valid topic from that ? Because if it doesn't reject then it must have identified a topic in the database, so we could just proceed straight onto the `selectArticles` model

Reply:
Yeah this might be a good long-term solution! Currently there is no selectTopic but once I build this endpoint it would be great here.